src/Impostor.Server/Net/Inner/Objects/InnerGameData.TaskInfo.Api.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Impostor.Api;
+using Impostor.Api.Net.Inner;
+using Impostor.Api.Net.Messages.Rpcs;
+using Impostor.Server.Events.Player;
+
+namespace Impostor.Server.Net.Inner.Objects
+{
+    internal partial class InnerGameData
+    {
+        public partial class TaskInfo
+        {
+            public async ValueTask CompleteAsync()
+            {
+                if (_playerInfo.Controller == null)
+                {
+                    throw new ImpostorException("Can't complete a task that doesn't have a player assigned");
+                }
+
+                var player = _playerInfo.Controller;
+
+                if (Complete)
+                {
+                    return;
```

Comment:
do we want to throw an exception when api users try to complete an already completed tasks? The method not doing anything when it is already completed would surprise me